cmd/ssh_key_register.go

```diff
@@ -43,35 +42,53 @@ func (c *computeSSHKeyRegisterCmd) cmdPreRun(cmd *cobra.Command, args []string)
 
 func (c *computeSSHKeyRegisterCmd) cmdRun(cmd *cobra.Command, _ []string) error {
 	var (
-		sshKey *egoscale.SSHKey
+		sshKey *v3.SSHKey
 		err    error
 	)
 
 	// Template registration can take a _long time_, raising
 	// the Exoscale API client timeout as a precaution.
-	globalstate.EgoscaleClient.SetTimeout(30 * time.Minute)
+	client := globalstate.EgoscaleV3Client.WithHttpClient(&http.Client{Timeout: 30 * time.Minute})
 
-	ctx := exoapi.WithEndpoint(
-		gContext,
-		exoapi.NewReqEndpoint(account.CurrentAccount.Environment, account.CurrentAccount.DefaultZone),
-	)
+	ctx := gContext
 
 	publicKey, err := os.ReadFile(c.PublicKeyFile)
 	if err != nil {
 		return err
 	}
 
-	decorateAsyncOperation(fmt.Sprintf("Registering SSH key %q...", c.Name), func() {
-		sshKey, err = globalstate.EgoscaleClient.RegisterSSHKey(ctx, account.CurrentAccount.DefaultZone, c.Name, string(publicKey))
+	registerKeyRequest := v3.RegisterSSHKeyRequest{
+		Name:      c.Name,
+		PublicKey: string(publicKey),
+	}
+
+	err = decorateAsyncOperations(fmt.Sprintf("Registering SSH key %q...", c.Name), func() error {
+		op, err := client.RegisterSSHKey(ctx, registerKeyRequest)
+
+		if err != nil {
+			return err
+		}
+
+		_, err = client.Wait(ctx, op, v3.OperationStateSuccess)
+		if err != nil {
+			return err
+		}
+
+		return nil
 	})
 	if err != nil {
 		return err
 	}
 
 	if !globalstate.Quiet {
+		sshKey, err = client.GetSSHKey(ctx, c.Name)
+		if err != nil {
+			return err
+		}
+
 		return c.outputFunc(&computeSSHKeyShowOutput{
-			Fingerprint: *sshKey.Fingerprint,
-			Name:        *sshKey.Name,
+			Fingerprint: sshKey.Fingerprint,
+			Name:        sshKey.Name,
 		}, nil)
```

Comment:
You can take inspiration of this create command:
https://github.com/exoscale/cli/blob/master/cmd/blockstorage_create.go#L90-94

and use it with `computeSSHKeyShowCmd`

```Golang
return (&computeSSHKeyShowCmd{
		Name: c.Name,
	}).cmdRun(nil, nil)
```

something like that